engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectColumnLayer.java

```diff
@@ -110,7 +110,13 @@ final public class SelectColumnLayer extends SelectOrViewColumnLayer {
 
     private ChunkSource<Values> getChunkSource() {
         if (chunkSource == null) {
-            chunkSource = ReinterpretUtils.maybeConvertToPrimitive(selectColumn.getDataView());
+            ColumnSource<?> dataSource = selectColumn.getDataView();
+            if (dataSource.getType() != writableSource.getType()) {
+                // this should only occur when using primitives internally and the user has requested a non-primitive
+                chunkSource = ReinterpretUtils.maybeConvertToPrimitive(dataSource);
+            } else {
```

Comment:
We should follow the reinterpret with an `Assert` that the resulting types are now equal.